src/main/scala/com.snowplowanalytics.snowplow.enrich.beam/Enrich.scala

```diff
@@ -264,18 +268,33 @@ object Enrich {
   private def enrich(
     data: Array[Byte],
     enrichmentRegistry: EnrichmentRegistry[Id],
-    client: Client[Id, Json]
+    client: Client[Id, Json],
+    sentryDSN: Option[String]
   ): List[Validated[BadRow, EnrichedEvent]] = {
     val processor = Processor(generated.BuildInfo.name, generated.BuildInfo.version)
     val collectorPayload = ThriftLoader.toCollectorPayload(data, processor)
-    EtlPipeline.processEvents(
-      new AdapterRegistry,
-      enrichmentRegistry,
-      client,
-      processor,
-      new DateTime(System.currentTimeMillis),
-      collectorPayload
-    )
+    Either.catchNonFatal(
+      EtlPipeline.processEvents(
+        new AdapterRegistry,
+        enrichmentRegistry,
+        client,
+        processor,
+        new DateTime(System.currentTimeMillis),
+        collectorPayload
+      )
+    ) match {
+      case Left(throwable) =>
+        logger.error(
+          s"Problem occured while processing CollectorPayload [$collectorPayload]",
+          throwable
+        )
+        sentryDSN.foreach { dsn =>
+          System.setProperty("sentry.dsn", dsn)
```

Comment:
Am I right that there's no way to set it once per job because it will be potentially running on multiple nodes?

Reply:
Indeed, because `SentryClient` (like we use in Stream Enrich) is not serializable